File: packages/guards/src/profanity-ru.ts
import type { Guard, GuardResult, GuardContext } from 'open-guardrail-core';

interface ProfanityRuOptions {
  action: 'block' | 'warn';
}

const WORDS: string[] = [
  'блядь', 'сука', 'хуй', 'пиздец', 'ебать',
  'мудак', 'дурак', 'козёл', 'дебил', 'тварь',
];

export function profanityRu(options: ProfanityRuOptions): Guard {
  return {
    name: 'profanity-ru',
    version: '0.1.0',
    description: 'Russian profanity detection',
    category: 'locale',
    supportedStages: ['input', 'output'],
    async check(text: string, _ctx: GuardContext): Promise<GuardResult> {
      const start = performance.now();
      const lower = text.toLowerCase();
      const matched: string[] = [];

      for (const word of WORDS) {
        if (lower.includes(word)) matched.push(word);
      }

      const triggered = matched.length > 0;

      return {
        guardName: 'profanity-ru',
        passed: !triggered,
        action: triggered ? options.action : 'allow',
        score: triggered ? Math.min(matched.length / 3, 1.0) : 0,
        latencyMs: Math.round(performance.now() - start),
        details: triggered ? { matched } : undefined,
      };
    },
  };
}

File: python/open_guardrail/guards/profanity_ru.py
"""Detect Russian profanity."""
from __future__ import annotations

import re
import time

from open_guardrail.core import GuardResult

_WORDS = [
    "блядь", "сука", "хуй", "пиздец",
    "ебать", "мудак", "дурак", "козёл",
    "дебил", "тварь",
]

_PATTERNS = [
    re.compile(rf"\b{w}\b", re.IGNORECASE) for w in _WORDS
]


class _ProfanityRu:
    def __init__(self, *, action: str = "block") -> None:
        self.name = "profanity-ru"
        self.action = action

    def check(
        self, text: str, stage: str = "input"
    ) -> GuardResult:
        start = time.perf_counter()
        matched: list[str] = []

        for pat in _PATTERNS:
            if pat.search(text):
                matched.append(pat.pattern)

        triggered = len(matched) > 0
        score = min(len(matched) / 3, 1.0) if triggered else 0.0
        elapsed = (time.perf_counter() - start) * 1000

        return GuardResult(
            guard_name="profanity-ru",
            passed=not triggered,
            action=self.action if triggered else "allow",
            score=score,
            message=(
                "Russian profanity detected"
                if triggered
                else None
            ),
            latency_ms=round(elapsed, 2),
            details=(
                {
                    "matched_count": len(matched),
                    "reason": (
                        "Text contains Russian profane"
                        " or offensive words"
                    ),
                }
                if triggered
                else None
            ),
        )


def profanity_ru(
    *, action: str = "block"
) -> _ProfanityRu:
    return _ProfanityRu(action=action)
